wordfreq_builder/wordfreq_builder/ninja.py

```diff
@@ -238,25 +238,42 @@ def jieba_deps(dirname_in, languages):
 
 def reddit_deps(dirname_in, languages):
     lines = []
-    if not languages:
-        return lines
-    assert languages == ['en']
-
-    processed_files = []
     path_in = pathlib.Path(dirname_in)
+    slices = {}
+    counts_by_language = defaultdict(list)
+
+    # Extract text from the Reddit comment dumps, and write them to
+    # .txt.gz files
     for filepath in path_in.glob('*/*.bz2'):
-        base = filepath.name[:-4]
-        transformed_file = wordlist_filename('reddit', 'en', base + '.txt.gz')
+        base = filepath.stem
+        transformed_file = wordlist_filename('reddit', base + '.all', 'txt')
+        slices[base] = transformed_file
         add_dep(lines, 'extract_reddit', str(filepath), transformed_file)
-        count_file = wordlist_filename('reddit', 'en', base + '.counts.txt')
-        add_dep(lines, 'count', transformed_file, count_file)
-        processed_files.append(count_file)
 
-    output_file = wordlist_filename('reddit', 'en', 'counts.txt')
-    add_dep(
-        lines, 'merge_counts', processed_files, output_file,
-        params={'cutoff': 3}
-    )
+    for base in sorted(slices):
+        transformed_file = slices[base]
+        language_outputs = []
+        for language in languages:
+            filename = wordlist_filename('reddit', base + '.' + language, 'txt')
+            language_outputs.append(filename)
+
+            count_filename = wordlist_filename('reddit', base + '.' + language, 'counts.txt')
+            add_dep(lines, 'count', filename, count_filename)
+            counts_by_language[language].append(count_filename)
+
+        # find the prefix by constructing a filename, then stripping off
+        # '.xx.txt' from the end
+        prefix = wordlist_filename('reddit', base + '.xx', 'txt')[:-7]
```

Comment:
This is all a bit of a workaround. `base` is the filename of the input file, which comes in many pieces broken up by date.

I had originally written something like `wordlist_filename('reddit-' + base, ...)` but the problem is that 'reddit' is a key that it uses to look up which directory to put this in. So I prepended `base` to the language code instead.

The `.xx` here is so that the filename is being constructed the same way as the other ones above. Yes, there is a shorter expression that would compute the same thing, but then it would be even less apparent why it's like the other expressions above.